src/Avalonia.Visuals/VisualExtensions.cs

```diff
@@ -34,48 +31,93 @@ public static Point PointToClient(this IVisual visual, PixelPoint point)
         /// <returns>The point in screen coordinates.</returns>
         public static PixelPoint PointToScreen(this IVisual visual, Point point)
         {
-            var p = GetRootAndPosition(visual);
-            return p.Item1.PointToScreen(point + p.Item2);
+            var p = visual.TranslatePoint(point, visual.VisualRoot);
```

Comment:
Can we sure the visual.VisualRoot can get the transform from TransformToVisual? If the visual.VisualRoot can not get it that the `p` may be null.

Reply:
Maybe the visual is IRenderRoot that can not find the common ancestor between visual and visual.VisualRoot.

Reply:
The default visual in Avalonia will return self when it is IRenderRoot.

https://github.com/AvaloniaUI/Avalonia/blob/b777bfa187d7dd5fca749453eba7957c1c9d27e9/src/Avalonia.Visuals/Visual.cs#L249